src/atoms/Space/Space.native.js

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components/native';
+
+import { makePxValues } from '../../_helpers/theme';
+
+const SpaceEnhancer = styled(({ margin, padding, children, ...props }) =>
+  React.cloneElement(children, props),
+)`
+  ${(props) => (props.margin ? `margin: ${makePxValues(props.margin)}` : '')};
+  ${(props) => (props.padding ? `padding: ${makePxValues(props.padding)}` : '')};
+`;
+
+const Space = ({ margin, padding, children, ...props }) => {
+  if (React.Children.toArray(children).length === 1) {
+    return <SpaceEnhancer margin={margin} padding={padding} children={children} {...props} />;
```

Comment:
why do you need to create a new component? also it should be be fail fast.
```jsx
if(React.Children.toArray(children).length !== 1){
	 throw new Error('Expected a single child for Space component');
}
return React.cloneElement(children, props)
```

Reply:
I had done the same thing using styled component, it was throwing error, but this snippet is working fine! @shubhnik lol.
Changes are done.